File: backend/app/models/product.py
from pydantic import Field

from app.models.common import CamelModel


class ProductReview(CamelModel):
    reviewer_name: str
    rating: int = Field(ge=1, le=5)
    comment: str
    date: str | None = None


class ProductSummary(CamelModel):
    id: str
    name: str
    price: float
    short_description: str
    thumbnail_url: str
    category: str


class Product(ProductSummary):
    long_description: str
    reviews: list[ProductReview] = []


class ProductListResponse(CamelModel):
    items: list[ProductSummary]
    total: int

File: frontend/src/types/product.ts
export interface ProductReview {
  reviewerName: string
  rating: number
  comment: string
  date?: string
}

export interface ProductSummary {
  id: string
  name: string
  price: number
  shortDescription: string
  thumbnailUrl: string
  category: string
}

export interface Product extends ProductSummary {
  longDescription: string
  reviews: ProductReview[]
}

export interface ProductListResponse {
  items: ProductSummary[]
  total: number
}

export type SortBy = 'price' | 'name'
export type SortOrder = 'asc' | 'desc'

export interface ProductFilters {
  name: string
  category: string
  sortBy: SortBy | ''
  sortOrder: SortOrder
}
